samples/same54xprov2/basic-sample/src/iotconnect_app.c

```diff
@@ -0,0 +1,310 @@
+    // TelemetryAddWith* calls are only required if sending multiple data points in one packet.
+    iotcl_telemetry_add_with_iso_time(msg, iotcl_iso_timestamp_now());
+    iotcl_telemetry_set_string(msg, "version", APP_VERSION);
+    // random number 0-100, cast to int so that it removes decimals in json
+    iotcl_telemetry_set_number(msg, "random", (int)((double)rand() / (double)RAND_MAX * 100.0));
+
+    
+#ifdef IOTCONNECT_APP_USE_WEATHER_CLICK
+    Weather_readSensors();
+    iotcl_telemetry_set_number(msg, "temperature", Weather_getTemperatureDegC());
+    iotcl_telemetry_set_number(msg, "pressure", Weather_getPressureKPa());
+    iotcl_telemetry_set_number(msg, "humidity", Weather_getHumidityRH());
+#endif    
+    const char *str = iotcl_create_serialized_string(msg, false);
+    iotcl_telemetry_destroy(msg);
+    printf("Sending: %s\r\n", str);
+    iotconnect_sdk_send_packet(str); // underlying code will report an error
+    iotcl_destroy_serialized(str);
+}
+
+/* Include the sample.  */
+bool iotconnect_sample_app(NX_IP *ip_ptr, NX_PACKET_POOL *pool_ptr, NX_DNS *dns_ptr) {
+    printf("Starting App Version %s\r\n", APP_VERSION);
+    IotConnectClientConfig *config = iotconnect_sdk_init_and_get_config();
+    azrtos_config.ip_ptr = ip_ptr;
+	azrtos_config.pool_ptr = pool_ptr;
+	azrtos_config.dns_ptr = dns_ptr;
+
+    config->cpid = IOTCONNECT_CPID;
+    config->env = IOTCONNECT_ENV;
+#ifdef IOTCONNECT_DUID
+    config->duid = IOTCONNECT_DUID; // custom device ID
+#else
+  #ifndef ENABLE_DDIM_PKCS11_ATCA_DRIVER_SAMPLE
+    config->duid = (char*) compose_device_id(); // <DUID_PREFIX>-<MAC Address>
+  #endif
+#endif
+    config->cmd_cb = on_command;
+    config->ota_cb = on_ota;
+    config->status_cb = on_connection_status;
+
+#ifdef IOTCONNECT_SYMETRIC_KEY
+    config->auth.type = IOTC_KEY;
+    config->auth.data.symmetric_key = IOTCONNECT_SYMETRIC_KEY;
+#else
+    config->auth.type = IOTC_X509;
+
+#ifdef ENABLE_DDIM_PKCS11_ATCA_DRIVER_SAMPLE
+    auth_driver_context = NULL;
+    struct pkcs11_atca_driver_parameters parameters = {0}; // dummy, for now
+    IotcDdimInterface ddim_interface;
+    IotcAuthInterfaceContext auth_context;
+    if(pkcs11_atca_create_auth_driver( //
+            &(config->auth.data.x509.auth_interface), //
+            &ddim_interface, //
+			&auth_context, //
+			&parameters)) { //
+    	return false;
+    }
+    config->auth.data.x509.auth_interface_context = auth_context;
+
+    uint8_t* cert;
+    size_t cert_size;
+
+    config->auth.data.x509.auth_interface.get_cert(
+		   	auth_context, //
+			&cert, //
+			&cert_size //
+			);
+    if (0 == cert_size) {
+        printf("Unable to get the certificate from the driver.\r\n");
+        pkcs11_atca_release_auth_driver(auth_context);
+        return false;
+    }
+    char* b64_string_buffer = malloc(1024); // cert should be around 500, so this should not exceed 1024
```

Comment:
Magic number

Reply:
If this is just a base64 encoded version or ASCII armored PEM version of the certificate then we can calculate the size at compile-time based on the size of the variable for holding the certificate